presto-spi/src/main/java/com/facebook/presto/spi/block/LazySliceArrayBlock.java

```diff
@@ -182,8 +183,13 @@ public void assureLoaded()
             throw new IllegalArgumentException("Lazy block loader did not load this block");
         }
 
-        if (dictionary && ids.length != positionCount) {
-            throw new IllegalArgumentException(format("Expected %s positions, loaded %s", positionCount, ids.length));
+        if (dictionary) {
+            if (ids.length != positionCount) {
+                throw new IllegalArgumentException(format("Expected %s positions, loaded %s", positionCount, ids.length));
```

Comment:
This should not be `IllegalArgumentException` since this method has no arguments. Is it always valid to call `assureLoaded()`? If not, and it can only be called at certain times, then `IllegalStateException` is appropriate. Otherwise, if this is caused by a bug, then it should be `RuntimeException`